File: examples/python/script_stats.py
"""Count languages by script type using the index."""
from __future__ import annotations

from collections import Counter

from worldalphabets import get_index_data


def main() -> None:
    index = get_index_data()
    counts = Counter(item["script-type"] for item in index)
    for script, count in counts.most_common():
        print(f"{script}: {count}")


if __name__ == "__main__":
    main()

File: examples/node/scriptStats.js
// Count languages by script type.
const { getIndexData } = require('../..');

async function main() {
  const index = await getIndexData();
  const counts = index.reduce((acc, item) => {
    const type = item['script-type'];
    acc[type] = (acc[type] || 0) + 1;
    return acc;
  }, {});
  for (const [type, count] of Object.entries(counts)) {
    console.log(`${type}: ${count}`);
  }
}

main();
